CMakeLists.txt

```diff
@@ -29,17 +26,16 @@ elseif (BLA_VENDOR MATCHES ATLAS)
     set(BLAS_LIBRARIES "${BLAS_LIBRARIES};${CBLAS_LIB}")
 endif()
 
-# include_directories(SYSTEM
-#   C:\\nuget_openblas\\OpenBLAS.0.2.14.1\\lib\\native\\include\\
-#   C:\\OpenBLAS.0.2.14.1\\lib\\native\\include\\)
-
-#include the subdirectory containing our libs
+# include the subdirectory containing our libs
 add_subdirectory(Hello)
 include_directories(Hello)
-#indicate the entry point for the executable
+# indicate the entry point for the executable
 add_executable(CMakeHelloWorld Hello HelloWorld.cpp)
 
+# BLAS libraries compile dynamically against threads
+LIST(APPEND CMAKE_THREAD_LIBS_INIT ${LM})
```

Comment:
`${LM}` is `-llibm -llibdl` (aka `/usr/lib/x86_64-linux-gnu/libm.so` and `/usr/lib/x86_64-linux-gnu/libdl.so`)
This is only needed by *gcc* compilers, so it should be inside an if statement like `if (CMAKE_C_COMPILER MATCHES ".+gcc")`

More over it should be discussed if [this `find_package`](https://github.com/massich/cpp-mkl-test/blob/576bf799b267edd9cdea915575d15438f7604ea1/cmake/FindBLAS.cmake#L500) should be done silently or not.

Advantages of doing it silently are that `target_link_libraries(CMakeHelloWorld Hello ${BLAS_LIBRARIES})` is clear. However to make BLA work with static needs to properly compose the thread libs which makes `target_link_libraries(CMakeHelloWorld Hello ${BLAS_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})` look like a hack. 

Maybe it would be better to just check that Threads is found at the begining of `FindBLA.cmake` and break if not found, like this is explicit what needs to be done with trheads, and allows for better control. ( In that case, `${LM}` should be placed where it's due, and it would make sense `Intel10_64lp` appending what needs inside `FindBLA.cmake`)

cc: @agramfort, @aabadie, @Kyc0o

Reply:
sounds a good way to clean it up indeed.